File: easy/12. Tandem Bicycle/tandem_bicycle.cpp
#include <algorithm>
#include <bits/stdc++.h>
#include <vector>

int tandemBicycle(std::vector<int> redShirtSpeeds,
                  std::vector<int> blueShirtSpeeds, bool fastest) {
  int result = 0;
  std::sort(redShirtSpeeds.begin(), redShirtSpeeds.end());
  if (fastest) {
    std::sort(blueShirtSpeeds.begin(), blueShirtSpeeds.end(),
              std::greater<int>());
  } else {
std::sort(blueShirtSpeeds.begin(), blueShirtSpeeds.end(),
              std::greater<int>());

  }

  for (int i = 0; i < redShirtSpeeds.size(); i++) {
    result += std::max(redShirtSpeeds[i], blueShirtSpeeds[i]);
  }

  return result;
}

File: easy/12. Tandem Bicycle/tandem_bicycle.go
package main

import (
	"sort"
)

func TandemBicycle(redShirtSpeeds []int, blueShirtSpeeds []int, fastest bool) int {
	result := 0

	sort.Ints(redShirtSpeeds)
	if fastest {
		sort.Sort(sort.Reverse(sort.IntSlice(blueShirtSpeeds)))
	} else {
		sort.Ints(blueShirtSpeeds)
	}

	for i := 0; i < len(redShirtSpeeds); i++ {
		rss := redShirtSpeeds[i]
		bss := blueShirtSpeeds[i]
		if rss > bss {
			result += rss
		} else {
			result += bss
		}
	}

	return result
}
